File: 2 semester/ISP/Laba 1 ISP - Matrix/DataStructures/LinkedListItem.pas
unit LinkedListItem;

interface

uses
      Object_;

type
      {* Внутреннее представление *}
      TLinkedListItem = class
      private
            _obj: TObject_;
            _next: TLinkedListItem;
            _prev: TLinkedListItem;
      public
            constructor Create(const obj: TObject_);
            destructor Destroy(); override;
      public
            property Obj: TObject_ read _obj write _obj;
            {* Следующий элемент *}
            property Next: TLinkedListItem read _next write _next;
            {* Предыдущий элемент *}
            property Prev: TLinkedListItem read _prev write _prev;
      end;

implementation

constructor TLinkedListItem.Create(const obj: TObject_);
begin
      _obj := obj;
      _next := nil;
      _prev := nil;
end;

destructor TLinkedListItem.Destroy();
begin
      if (_next <> nil) then
      begin
            if (_next <> Self) then
            begin
                  _next.Free();
            end;
            _next := nil;
            _prev := nil;
      end;
      inherited;
end;

end.

File: 2 semester/ISP/Laba 6 ISP - Set/Laba 6 ISP - Set/LinkedListItem.cs
namespace Laba_6_ISP___Set
{
    public class LinkedListItem<T>
    {
        public LinkedListItem<T> Next { get; set; }

        public LinkedListItem<T> Prev { get; set; }

        public T Value { get; set; }

        public LinkedListItem(T value)
        {
            Value = value;
        }

        public override string ToString()
        {
            return Value.ToString();
        }
    }
}
